core/commands/add.go

```diff
@@ -113,12 +113,16 @@ remains to be implemented.
 					return
 				}
 
-				err = n.Pinning.Pin(context.Background(), rootnd, true)
+				rnk, err := rootnd.Key()
 				if err != nil {
 					res.SetError(err, cmds.ErrNormal)
 					return
 				}
 
+				mp := n.Pinning.GetManual()
+				mp.RemovePinWithMode(rnk, pin.Indirect)
```

Comment:
> is the worry that something will GC in-between, while adding? (if so, ok makes sense).

This too, a poorly timed gc could end really badly for us. (although its still not perfectly safe)